### PR TITLE
add some pronouns nouns and abbreviations

### PR DESCRIPTION
**Pull request by phoet**
do the words need to be lower-case for the normalization process?

i'd like to keep them in upper-case (the proper way of writing them) and then maybe to-lower-case them in the import process?

what i did here was basically to quickly look for some word lists. there are probably much better lists out there, i found some with hundred thousands of words. what would be an acceptable size for the lib to handle?

### CODE REVIEW

File: src/data/abbreviations.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ let compact = {
   ],
 
   Place: [
+    'Str',
```
Comment (phoet) [inline src/data/abbreviations.js]: this would usually be abbreviated with `Str.` would the dot be normalized away by compromise?

Comment (spencermountain) [inline src/data/abbreviations.js]: yeah, those abbreviations are used in the sentence-splitter, so it doesn't chop-it-up there

**Comment by spencermountain** [comment]
YES YES YES! very nice!
yeah, so in the english one we heavily-normalize each input, so that all the internal stuff is easier and less picky - are there different rules about capitalization in german? or punctuation? The european comma/decimal number format is fine, and we can do that no problem.

if you can anticipate any tricky stuff like this, lemme know em, otherwise i'm heading straight into them ;)

yeah, en-compromise is able to be so small because we conjugate all the verb forms, and pluralize all the nouns. Those are the two big wins, if you wanna look at those maybe?

i'm gonna poke-around at one of the german wordnets, and see if i can re-purpose the [word-classifier](https://github.com/nlp-compromise/thumb) i used in -en
do you use slack?
cheers!

**Comment by spencermountain** [comment]
yeah, what would be best is if you looked at ways to expand any root words into more forms- it doesn't need to be fancy, at all:
* student -> studenten
* versammeln -> Versammelt
* kalt -> kälteste
this gives us ways to multiply our known word-lists, so the library can stay web-size.

oh - i don't know what your opinion on this is. right now it normalizes unicode to ascii internally, so `ä -> a`. I know this is grotesque, but its for trying to support modern, causual internet-speak, if someone types `Es ist am kaltesten`, it would still understand it.  - do people lazily use ascii in german, like in english?

**Comment by spencermountain** [comment]
oh wow, nouns are all capitalized.
yeah, that helps us a lot. Let's keep them lowercase in the lexicon, but the tagger can do that pass easily.
holy, that's nice!

**Comment by phoet** [comment]
ah ok, i'll try to understand what the lib can do so that the list of words will stay at a minimum.

i think german is actually pretty good in terms of sticking to rules. ie this is a set of rules to determine if a given noun is male, female or neutral http://kommunikationsabc.de/2013/04/09/tabelle-substantive-endungen-und-die-passenden-artikel/

**Comment by spencermountain** [comment]
hey, any objections to doing this internally?
image
it wouldn't change any of the actual text. youtube says it's common, though what do i know

**Comment by phoet** [comment]
@spencermountain in general, i think that in german you use umlauts quite regularly, even when using mobile phones etc. auto-suggest and completion do have it automatically.

for stemming, i think that you would substitute umlauts with their long form `ae` etc. 

i think for this purpose it's ok to completely drop them. i think there are only a few edgecases where replacing umlauts with their base would create a duplicate word or something like this.

so my suggestion would be to `ä => a` and to maybe even `ae => a`. a whitelist could make sure to catch words were those combinations are appropriate (found a [list of words](http://wortsuche.com/contains-by-length/ae/), but it's really rare ones) 

some is true for nouns. i think that in most cases you can safely ignore the uppercase letters (i use to write german downcase most of the time).

**Comment by spencermountain** [comment]
rebel!
ah, interesting. yeah, i don't mind keeping the umlauts in the lexicon, the only issue would be when people don't use them, they'd miss the word. it's probably less error-prone going from `ä => a` than backward.

did a bunch of work yesterday, but it's not really sensible yet.

the capital thing is a huge deal. That makes our job so much easier, but I agree we ought to support  downcase, or lazy, casual-writing. I clicked around in twitter and reddit and people seem to be titlecasing always. I don't know what conditions, or false-positives there will be.

the question I have is this - when we don't know a word in english, compromise assumes it's a noun. If we can see that the sentence is being titlecased, and we don't know a lowercase word, does that mean we should assume it's not a noun? what do you think?

so yesterday i got a [list of the most-common infinitives](https://github.com/nlp-compromise/de-compromise/blob/master/src/tagger/lexicon/infinitives.json), if you wanted to take a crack at conjugating them?
cheers

**Comment by spencermountain** [comment]
ugh, I didn't know adjectives have genders too. uber-müll!

**Comment by spencermountain** [comment]
[check it out!](https://rawgit.com/nlp-compromise/de-compromise/master/demo/basic/index.html)

**Comment by phoet** [comment]
looks like you are excited about this :-D 

i habe to take care of family business and come back to you as quickly as possible!

**Pull request merged**